### PR TITLE
[node-manager] Don't deploy standby-holder if there is minPerZone == maxPerZone

### LINKED ISSUE #214: [node-manager][standby-holder] Holder is being deployed if minPerZone == maxPerZone
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.24.17

### Expected Behavior

We don't need standby-holder if minPerZone and maxPerZone are equal.

### Actual Behavior

standby-holder is always being deployed.

### Steps To Reproduce

_No response_

### Additional Information

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by apolovov**
## Description
Don't deploy standby-holder if there is minPerZone == maxPerZone. Closes #214.

## Why do we need it, and what problem does it solve?
In some cases there is useless deployment in Pending state which doesn't have chances to be deployed.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Don't deploy standby-holder if there is minPerZone == maxPerZone.
impact_level: low
```

### CODE REVIEW

File: modules/040-node-manager/hooks/discover_standby_ng.go [hunk 1]
```diff
@@ -65,8 +61,22 @@ func standbyNodeGroupFilter(obj *unstructured.Unstructured) (go_hook.FilterResul
 		taints = make([]v1.Taint, 0)
 	}
 
+	needStandby := false
+	maxPerZone := 0
```
Comment (apolovov) [inline modules/040-node-manager/hooks/discover_standby_ng.go:65]: It was "1" befor my PR. It confuses me.

File: modules/040-node-manager/hooks/discover_standby_ng.go [hunk 2]
```diff
@@ -65,8 +61,22 @@ func standbyNodeGroupFilter(obj *unstructured.Unstructured) (go_hook.FilterResul
...
+	if nodeGroup.Spec.NodeType == ngv1.NodeTypeCloudEphemeral {
+		maxPerZone = int(*nodeGroup.Spec.CloudInstances.MaxPerZone)
```
Comment (diafour) [inline modules/040-node-manager/hooks/discover_standby_ng.go:67]: > runtime error: invalid memory address or nil pointer dereference

The same is below for MinPerZone.

Comment (apolovov) [reply modules/040-node-manager/hooks/discover_standby_ng.go:67]: I rely on openapi validation. 
minPerZone/maxPerZone are mandatory for CloudEphemeral NGs.

Comment (diafour) [inline modules/040-node-manager/hooks/discover_standby_ng.go:67]:
```suggestion
		// No nil-checking for MaxPerZone and MinPerZone pointers as these fields are mandatory for CloudEphemeral NGs.
		maxPerZone = int(*nodeGroup.Spec.CloudInstances.MaxPerZone)
```

**Review by diafour — APPROVED** [review]
Looks good, only one suggestion left.

**Review by diafour — APPROVED** [review]

**Review by nabokihms — CHANGES_REQUESTED** [review]
Rebasing is required.

**Pull request merged**